Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM docker.io/library/node:lts-alpine AS base
+
+# Prepare work directory
+WORKDIR /elk
+
+FROM base AS builder
+
+# Prepare pnpm ( refer to https://pnpm.io/installation#on-alpine-linux )
+RUN wget -qO /bin/pnpm "https://github.com/pnpm/pnpm/releases/latest/download/pnpm-linuxstatic-x64" && chmod +x /bin/pnpm
+
+# Prepare deps
+RUN apk update
+RUN apk add git --no-cache
+
+# Copy all files
+COPY . ./
+
+# Build
+RUN pnpm i --frozen-lockfile
```

Comment:
It might be better to first copy the lock files and run the install. Then you copy in the remaining files and build. This improves the caching performance on any potential CI as it should not have to download all the dependencies for every run (just ones in which the lock files change)